lib/modules/platform/gitlab/index.ts

```diff
@@ -142,10 +143,23 @@ export async function initPlatform({
 }
 
 // Get all repositories that the user has access to
-export async function getRepos(): Promise<string[]> {
+export async function getRepos(config?: AutodiscoverConfig): Promise<string[]> {
   logger.debug('Autodiscovering GitLab repositories');
+
+  let topicString = '';
+  if (config?.topics?.length) {
+    topicString = `&topic=${encodeURIComponent(config.topics.join(','))}`;
+  }
+
   try {
-    const url = `projects?membership=true&per_page=100&with_merge_requests_enabled=true&min_access_level=30&archived=false`;
+    const url =
+      'projects' +
+      '?membership=true' +
```

Comment:
we should probably use the urlparams class here instead. it makes param handling simpler

Reply:
Oh, much better; thanks.